### PR TITLE
Feature/movie

### PR DESCRIPTION
**Pull request by aspitz1**
#3 #15 
* Adds fetch of movie upon clicking on a movie cover
* replaces previous `setState` in the `viewMovie` method in App with `setState` inside of a fetch causing a page update
* Functions but has not styled

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import Navbar from '../Navbar/Navbar';
 import Movie from '../Movie/Movie';
 import AllMovies from '../All-Movies/All-Movies';
 import movieData from '../movieData';
+import { getMovie } from '../api-calls/apiCalls'
```
Comment (aspitz1) [inline src/App/App.js:7]: I am exporting the `getMovies` method as a object, using destructuring here

File: src/App/App.js [hunk 2]
```diff
@@ -36,8 +33,24 @@ class App extends Component {
     this.setState({ ...this.state, isMovieView: false });
   }
  viewMovie = (id) => {
-    console.log('click')
-    this.setState({...this.state, isMovieView: true, movie:{...this.state.movie, id: id}});
+    getMovie(id)
```
Comment (aspitz1) [inline src/App/App.js]: Passing the `id` as the endpoint ended up being the move vs updating the id. When the `id` was updated it reloaded the components and messed up the process.

File: src/App/App.js [hunk 3]
```diff
@@ -36,8 +33,24 @@ class App extends Component {
...
+          posterPath: data.movie['poster_path'],
+          backdropPath: data.movie['backdrop_path'],
+          releaseDate: data.movie['release_date'],
+          overview: data.movie.overview,
+          genres: data.movie.genres,
+          budget: data.movie.budget,
+          revenue: data.movie.revenue,
+          runtime: data.movie.runtime,
+          tagline: data.movie.tagline,
+          averageRating: data.movie['average_rating']
```
Comment (aspitz1) [inline src/App/App.js]: The way the data comes back bracket notation is needed to account for the underscore in the key

File: src/Movie/Movie.js [hunk 4]
```diff
@@ -13,20 +13,20 @@ const Movie = ({ movieDetails }) => {
...
+                    <li>Average Rating: {averageRating.toFixed(2)}</li>
+                    <li>{genres.join(' | ')} Released: {releaseDateFormatted}</li>
                     <li>{runtime} Minutes</li>
                     <li>Budget: ${budget.toLocaleString('en-US')}</li>
                     <li>Revenue: ${revenue.toLocaleString('en-US')}</li>
                 </ul>
             </div>
             <div className="movie-description-container">
                 <div>
-                    {tagline.length && <h3 className="tagline">{tagline}</h3>}
```
Comment (aspitz1) [inline src/Movie/Movie.js:29]: This was printing '0' when there was no tagline lol so using the falsy value of an empty string works better

File: src/App/App.js (hunk 1 above)
Comment (mayakappen) [reply src/App/App.js:7]: whoa cool!

File: src/api-calls/apiCalls.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const getMovie = (endpoint) => {
+    return fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${endpoint}`)
+}
+
+export { getMovie }
```
Comment (mayakappen) [inline src/api-calls/apiCalls.js:5]: love how you make this dynamic!

File: src/App/App.js [hunk 6]
```diff
@@ -11,34 +12,47 @@ class App extends Component {
...
+            overview: data.movie.overview,
+            genres: data.movie.genres,
+            budget: data.movie.budget,
+            revenue: data.movie.revenue,
+            runtime: data.movie.runtime,
+            tagline: data.movie.tagline,
+            averageRating: data.movie['average_rating']
+          }})
+        })
+  }
```
Comment (mayakappen) [inline src/App/App.js:55]: very cool! I love how you used the getMovie method here.

File: src/Movie/Movie.js (hunk 4 above)
Comment (mayakappen) [reply src/Movie/Movie.js:29]: nice work looks great!

**Review by mayakappen — COMMENTED** [review]
Very cool, do you want me to put my fetch in that same apiCalls file after I merge it?

**Comment by aspitz1** [comment]
@mayakappen Yeah! feel free to use the same API file. You can export the function inside the same object, using commas to separate and destructuring in the App.js.

Thank you!!!!

File: src/App/App.js (hunk 1 above)
Comment (aspitz1) [reply src/App/App.js:7]: > @mayakappen Yeah! feel free to use the same API file. You can export the function inside the same object, using commas to separate and destructuring in the App.js.
> 
> Thank you!!!!

At least I think this is how it works.

**Pull request merged**